### PR TITLE
React/week3

### PR DESCRIPTION
**Pull request by qamarfj**

### CODE REVIEW

File: react/week3/todos/src/App.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import "./App.css";
+import { Timer } from "./Timer";
+import Todos from "./Todos";
```
Comment (obelitz) [inline react/week3/todos/src/App.js:3]: It seems like you named the component file `todos.js` but you're importing `Todos.js` here

File: react/week3/todos/src/Border.js [hunk 2]
```diff
@@ -0,0 +1,3 @@
+export default function Border(props) {
+  return <div className="border">{props.children}</div>;
```
Comment (obelitz) [inline react/week3/todos/src/Border.js]: You use PascalCase for your other component classes, so I'd do the same thing here: `className="Border"`

File: react/week3/todos/src/TodosList.js [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+function TodoItem({ todo, DeleteTodo, UpdateTodo }) {
+  const [isCheked, setIsCheked] = useState(false);
+  const [isUpdated, setIsUpdated] = useState(true);
+  const [todoDescription, settodoDescription] = useState(todo.description);
+  const updateDescription = (e) => {
+    settodoDescription(e.target.value);
+  };
+  return (
+    <li className={isCheked ? "Done" : "notDone"}>
```
Comment (obelitz) [inline react/week3/todos/src/TodosList.js]: You are not using the `notDone` for currently, so you could use an empty string:
```suggestion
    <li className={isCheked ? "Done" : ""}>
```
or use `&&`

File: react/week3/todos/src/TodosList.js [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+          ))}
+        </ul>
+      )}
+      {todos.length < 1 && <div> No Items</div>}
+    </>
+  );
+}
+
+TodoItem.propTypes = {
+  todo: PropTypes.object,
```
Comment (obelitz) [inline react/week3/todos/src/TodosList.js]: If you know the values you're getting in an object, you can go even further and use the proptype `shape`. 
```
todo: PropTypes.shape({
  prop: PropTypes.whateverType,
  someOtherProp: PropTypes.someOtherType
})
```
See more here: https://reactjs.org/docs/typechecking-with-proptypes.html#proptypes

File: react/week3/todos/src/todos.js [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+  const [description, setDescription] = useState("");
+  const [deadline, setDeadline] = useState("");
+
+  useEffect(() => {
+    const fetchAPI = () => {
+      fetch(URL)
+        .then((response) => response.json())
+        .then((todos) => {
+          setIsLoaded((prev) => !prev);
+          setCurrentTodos((prev) => prev.concat(todos));
```
Comment (obelitz) [inline react/week3/todos/src/todos.js]: Nice with using the `concat` here

File: react/week3/todos/src/todos.js [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+    };
+    fetchAPI();
+    return () => {
+      setIsLoaded(() => {
+        return false;
+      });
+    };
+  }, []);
+
+  function AddTodo(e) {
```
Comment (obelitz) [inline react/week3/todos/src/todos.js]: I'd use camelCase for these three functions

File: react/week3/todos/src/todos.js [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+    setCurrentTodos((prev) => [...prev, currentTodo]);
+
+    setDescription("");
+    setDeadline("");
+
+    e.preventDefault();
+  }
+  const DeleteTodo = (id) => {
+    const todos = [...currentTodos];
+    setCurrentTodos(todos.filter((todo) => todo.id !== id));
```
Comment (obelitz) [inline react/week3/todos/src/todos.js]: Nice use of the `filter` method

File: react/week3/todos/src/todos.js [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+  };
+
+  function minDate() {
+    const now = new Date();
+    const month =
+      now.getMonth() > 9 ? now.getMonth() + 1 : "0" + (now.getMonth() + 1);
+    const day = now.getDate() > 9 ? now.getDate() : "0" + now.getDate();
+    const currentDate = "" + now.getFullYear() + "-" + month + "-" + day;
+    return currentDate;
+  }
```
Comment (obelitz) [inline react/week3/todos/src/todos.js]: Great thinking with this validation.

If you want, you can use backticks \` for string interpolation here:
```
const currentDate = `${now.getFullYear()}-${month}-${day}`
```

**Review by obelitz — CHANGES_REQUESTED** [review]
Nice! 👍 
You have that one `todos` import that fails when I run the code.

File: react/week3/todos/src/App.js (hunk 1 above)
Comment (qamarfj) [reply react/week3/todos/src/App.js:3]: Thank you for your valuable feedback and precious time you have putt on to review my home work. I have fixed the file name.

**Pull request merged**